sympy/combinatorics/free_groups.py

```diff
@@ -878,9 +930,18 @@ def is_dependent(self, word):
         is_independent
 
         """
-        self_st = str(self.letter_form)[1: -1]
-        return str(word.letter_form)[1: -1] in self_st or \
-                str((word**-1).letter_form)[1: -1] in self_st
+        i = None
+        try:
+            i = self.subword_index(word)
+        except:
```

Comment:
These lines could probably be simplified. For example,
```
try:
    self.subword_index(word)
    return True
except ValueError:
    pass
try:
    self.subword_index(word**-1)
    return True
except ValueError:
    return False
 ```
(I personally attempt to avoid nested try blocks though they are technically correct.)